### PR TITLE
#1792 - added user type checks to prevent guests from editing secure settings

### LINKED ISSUE #1792: Guests Should Not be able to Edit Their Secure Settings
### Description

We want guests to have a 'View Only' experience in FinishLine. This is already enforced in our backend, but isn't as obvious in some places in the frontend. This ticket is to hide the edit button on the user secure settings page block on the settings page if the current user is a guest.

### Acceptance Criteria

- [ ] The edit button for user secure settings is hidden if the current user is a guest

### Proposed Solution

See how we disable other buttons depending on the user's current role other places in the frontend for examples

### Mocks

N/A

### PR DESCRIPTION
**Pull request by matthewegg**
## Changes

Added user type checking to prevent guests from editing their secure settings.

## Notes

None

## Test Cases

Compared an admin, lead, team member and guest to make sure that only guests were unable to edit their settings.

## Screenshots
Image of admin logged in to FinishLine
image

Image of Admin User (guest) logged in to FinishLine
image

## To Do

Review and receive feedback

## Checklist

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [ ] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #1792

### CODE REVIEW

File: src/frontend/src/pages/SettingsPage/UserSecureSettings/UserSecureSettings.tsx [hunk 1]
```diff
@@ -31,14 +33,20 @@ export interface SecureSettingsFormInput {
 }
 
 const UserSecureSettings: React.FC<SecureSettingsProps> = ({ currentSettings }) => {
-  const [edit, setEdit] = useState(false);
+  const auth: Auth = useAuth();
```
Comment (RChandler234) [inline src/frontend/src/pages/SettingsPage/UserSecureSettings/UserSecureSettings.tsx:36]: you should use useCurrentUser instead

File: src/frontend/src/pages/SettingsPage/UserSecureSettings/UserSecureSettings.tsx [hunk 2]
```diff
@@ -31,14 +33,20 @@ export interface SecureSettingsFormInput {
 }
 
 const UserSecureSettings: React.FC<SecureSettingsProps> = ({ currentSettings }) => {
-  const [edit, setEdit] = useState(false);
```
Comment (RChandler234) [inline src/frontend/src/pages/SettingsPage/UserSecureSettings/UserSecureSettings.tsx:34]: you shouldn't need to move this line

File: src/frontend/src/pages/SettingsPage/UserSecureSettings/UserSecureSettings.tsx [hunk 3]
```diff
@@ -66,11 +74,11 @@ const UserSecureSettings: React.FC<SecureSettingsProps> = ({ currentSettings })
     <PageBlock
       title="User Secure Settings"
       headerRight={
-        !edit ? (
+        !edit && user.role !== 'GUEST' ? (
```
Comment (RChandler234) [inline src/frontend/src/pages/SettingsPage/UserSecureSettings/UserSecureSettings.tsx:77]: you should be able to use the isGuest method from shared here instead

**Review by RChandler234 — CHANGES_REQUESTED** [review]

**Comment by matthewegg** [comment]
got it. thanks for the feedback. i placed the line from the current develop branch back in its location, and changed the user identification to use useCurrentUser() and isGuest().

File: src/frontend/src/pages/SettingsPage/UserSecureSettings/UserSecureSettings.tsx [hunk 4]
```diff
@@ -66,11 +71,11 @@ const UserSecureSettings: React.FC<SecureSettingsProps> = ({ currentSettings })
     <PageBlock
       title="User Secure Settings"
       headerRight={
-        !edit ? (
+        !edit && !isGuest(user.role) ? (
```
Comment (RChandler234) [inline src/frontend/src/pages/SettingsPage/UserSecureSettings/UserSecureSettings.tsx:74]: you should be able to do one isGuest check for this whole statement: `!isGuest(user.role) && (is edit ? :)`
The && returns the right side of the operator only if the left side is true

**Review by RChandler234 — CHANGES_REQUESTED** [review]

File: src/frontend/src/pages/SettingsPage/UserSecureSettings/UserSecureSettings.tsx [hunk 5]
```diff
@@ -82,17 +88,17 @@ const UserSecureSettings: React.FC<SecureSettingsProps> = ({ currentSettings })
...
       <Grid container>
         {!edit ? (
           <UserSecureSettingsView settings={currentSettings} />
         ) : (
-          <UserSecureSettingsEdit currentSettings={currentSettings} onSubmit={handleConfirm} />
+          !isGuest(user.role) && <UserSecureSettingsEdit currentSettings={currentSettings} onSubmit={handleConfirm} />
         )}
       </Grid>
-      {edit && (
+      {!isGuest(user.role) && edit && (
```
Comment (RChandler234) [inline src/frontend/src/pages/SettingsPage/UserSecureSettings/UserSecureSettings.tsx:101]: you don't need this check or the check directly above. If a user can't click the edit button, then they won't be able to see either of these views anyways

**Review by RChandler234 — CHANGES_REQUESTED** [review]

File: src/frontend/src/pages/SettingsPage/UserSecureSettings/UserSecureSettings.tsx (hunk 5 above)
Comment (matthewegg) [reply src/frontend/src/pages/SettingsPage/UserSecureSettings/UserSecureSettings.tsx]: got it. i was having issues with the edit and cancel button rendering if the user was a guest, so i added the conditions, but this works.

**Review by RChandler234 — APPROVED** [review]
LGTM 🔥

**Pull request merged**